flink-clients/src/main/java/org/apache/flink/client/program/rest/RestClusterClient.java

```diff
@@ -168,13 +170,24 @@
                             .isPresent();
 
     public RestClusterClient(Configuration config, T clusterId) throws Exception {
-        this(config, clusterId, HighAvailabilityServicesUtils.createClientHAService(config));
+        this(
+                config,
+                clusterId,
+                new ClientHighAvailabilityServicesFactory() {
+                    @Override
+                    public ClientHighAvailabilityServices create(
+                            Configuration configuration, FatalErrorHandler fatalErrorHandler)
+                            throws Exception {
+                        return HighAvailabilityServicesUtils.createClientHAService(
+                                configuration, fatalErrorHandler);
+                    }
```

Comment:
Let's create a `DefaultClientHighAvailabilityServicesFactory` that has a singleton `INSTANCE` that does exactly this here. Then we don't have to use anonymous classes.